### PR TITLE
Support mpd001 by adding support for private git repos

### PR DESCRIPTION
**Pull request by Callek**

### CODE REVIEW

File: scriptworker/cot/verify.py [hunk 1]
```diff
@@ -2072,6 +2093,9 @@ def format(self, record):
             'verify_cot_signature': opts.verify_sigs,
         })
         context.config = apply_product_config(context.config)
+        if os.environ.get('SCRIPTWORKER_GITHUB_OAUTH_TOKEN'):
+            context.config['github_oauth_token'] = \
+                os.environ.get('SCRIPTWORKER_GITHUB_OAUTH_TOKEN')
```
Comment (escapewindow) [inline scriptworker/cot/verify.py]: This could go in `scriptworker.constants` (see `worker_id`, f/e).

File: scriptworker/cot/verify.py [hunk 2]
```diff
@@ -1879,6 +1896,10 @@ def verify_repo_matches_url(repo, url):
         bool: ``True`` if the repo matches the url.
 
     """
+    # Translate git@github.com:... to a path:
+    repo = repo.replace('git@github.com:', 'ssh://github.com/', 1)
+    if repo.endswith('.git'):
+        repo = repo[:-4]
```
Comment (escapewindow) [inline scriptworker/cot/verify.py]: We could create a helper function to do these things for us (tweak `get_source_repo` ?) but I'm ok with it like this.

File: scriptworker/github.py [hunk 3]
```diff
@@ -140,7 +140,7 @@ def is_github_url(url):
 
     """
     if isinstance(url, str):
-        return url.startswith('https://github.com/')
+        return url.startswith('https://github.com/') or url.startswith('ssh://github.com/')
```
Comment (escapewindow) [inline scriptworker/github.py]: I'm ok with this. We could also urlsplit and check for github.com in the domain.

**Comment by escapewindow** [comment]
Once we get mpd001 indexes set up, we may want to add mpd001 repos to the [list of indexes to verify cot against](https://github.com/mozilla-releng/scriptworker/blob/master/scriptworker/test/test_production.py#L64-L136) -- just to make sure any changes don't break mpd001.

File: scriptworker/constants.py [hunk 4]
```diff
@@ -337,6 +359,10 @@
                 'project:mobile:firefox-tv:releng:googleplay:product:firefox-tv': 'firefox-tv-repo',
                 'project:mobile:firefox-tv:releng:signing:cert:production-signing': 'firefox-tv-repo',
             }),
+            'mpd001': frozendict({
+                'project:releng:signing:cert:nightly-signing': 'mpd001-repo',
+                'project:releng:signing:cert:release-signing': 'mpd001-repo',
```
Comment (escapewindow) [inline scriptworker/constants.py]: I changed these to `project:mpd001:releng:signing:cert:...`

**Comment by Callek** [comment]
@escapewindow this is ready for review (with coverage) -- this is also unrebased though....

File: scriptworker/cot/verify.py [hunk 5]
```diff
@@ -1234,11 +1238,15 @@ def verify_link_in_task_graph(chain, decision_link, task_link):
         'event': {
             'after': commit_hash,
             'pusher': {
-                'email': committer_email,
+                # Github was returning lowercase for ${event.pusher.email} for
+                # an intercaps e-mail, but we could not find any documentation
+                # for why.
+                'email': committer_email.lower(),
```
Comment (escapewindow) [inline scriptworker/cot/verify.py]: This will change when I land my PR, but I think this works for now.

File: scriptworker/github.py [hunk 6]
```diff
@@ -113,6 +116,15 @@ def get_tag_hash(self, tag_name):
...
         repo = self._github_repository.html_url
+        if any(
+            vcs_rule.get('require_secret')
+            for vcs_rule in context.config['trusted_vcs_rules']
+        ):
+            # This check uses unofficial API on github, which we can't easily
+            # check for private repos, assume its true in the private case.
+            log.info("has_commit_landed_on_repository() not implemented for private"
+                     "repositories, assume True")
+            return True
```
Comment (escapewindow) [inline scriptworker/github.py]: We could assume that `git@` or `ssh://` imply `require_secret`, but I'm ok explicitly flagging it.

File: scriptworker/test/data/cotv2/private_github_.taskcluster.yml [hunk 7]
```diff
@@ -0,0 +1,294 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
```
Comment (escapewindow) [inline scriptworker/test/data/cotv2/private_github_.taskcluster.yml]: This file should probably go into `cotv4`, which is the latest data. We might drop the `cotv2` test data directory when we drop non-action-hook support.

File: scriptworker/test/test_production.py [hunk 8]
```diff
@@ -128,11 +128,11 @@ def build_config(override, basedir):
...
-}, {
-    "name": "focus nightly",
-    "index": "project.mobile.focus.signed-nightly.nightly.latest",
-    "task_type": "signing",
-    "cot_product": "mobile",
+# }, {
+#     "name": "focus nightly",
+#     "index": "project.mobile.focus.signed-nightly.nightly.latest",
+#     "task_type": "signing",
+#     "cot_product": "mobile",
```
Comment (escapewindow) [inline scriptworker/test/test_production.py]: We should either uncomment or remove completely, per channel.

**Pull request merged**